pkg/controllers/cspc-controller/sync.go

```diff
@@ -261,3 +275,137 @@ func (c *Controller) GetCSPIWithoutDeployment(cspc *cstor.CStorPoolCluster) ([]c
+	}
+	if cspi.Spec.PoolConfig.AuxResources == nil {
+		cspi.Spec.PoolConfig.AuxResources = cspc.Spec.DefaultAuxResources
+	}
+	if cspi.Spec.PoolConfig.Tolerations == nil {
+		cspi.Spec.PoolConfig.Tolerations = cspc.Spec.Tolerations
+	}
+
+	if cspi.Spec.PoolConfig.PriorityClassName == nil {
+		priorityClassName := cspc.Spec.DefaultPriorityClassName
+		cspi.Spec.PoolConfig.PriorityClassName = &priorityClassName
+	}
+}
+
+/*
+patchPoolDeploymentSpec calculates the diff (let us call this as 2-way patch data) between current existing pool
+manager deployment and desired existing pool manager.
+(The spec for desired pool manager is calculated/given/ordered by cspc-operator.)
+Once the 2-way patch data is calculated -- a strategic merge patch data is obtained by passing the current pool
+manager spec and 2-way patch data.
+Finally, a strategic merge patch is applied by using the strategic merge patch data.
+To understand more on how strategic merge patch works refer following doc:
+https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/strategic-merge-patch.md
+
+ToDo: Explore server side apply -- right now server side apply is still worked upon in k8s community.
+
+NOTE : A strategic merge patch happens only to specific fields in a certain native k8s object.
+For example, fields like `tolerations` in a deployment object cannot have strategic merge patch and are always
+a JSON Merge Patch ( RFC 6902 )
+
+*/
+func (pc *PoolConfig) patchPoolDeploymentSpec(cspi *cstor.CStorPoolInstance) error {
+	// Get the corresponding deployment for the cspi
+
+	existingDeployObj, err := pc.Controller.kubeclientset.AppsV1().Deployments(cspi.Namespace).Get(cspi.Name, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to get corresponding pool manager deployment for cspi %s in namespace %s", cspi.Name, cspi.Name)
+	}
+
+	newDeployObj := pc.AlgorithmConfig.GetPoolDeploySpec(cspi)
+
+	existingDeployObjInBytes, err := json.Marshal(existingDeployObj)
+
+	if err != nil {
+		return errors.Wrapf(err, "failed to marshal existing deployment object %s", existingDeployObj.Name)
+	}
+
+	newDeployObjInBytes, err := json.Marshal(newDeployObj)
+
+	if err != nil {
+		return errors.Wrapf(err, "failed to marshal new deployment object for existing deployment %s", existingDeployObj.Name)
+	}
+
+	twoWayPatchData, err := strategicpatch.CreateTwoWayMergePatch(existingDeployObjInBytes, newDeployObjInBytes, v1.Deployment{}, []mergepatch.PreconditionFunc{}...)
+	if err != nil {
+		return errors.Wrap(err, "could not compute two way patch data")
+	}
+
+	strategicPatchData, err := strategicpatch.StrategicMergePatch(existingDeployObjInBytes, twoWayPatchData, v1.Deployment{})
+	if err != nil {
+		return errors.Wrap(err, "could not compute strategic patch data")
+	}
+
+	_, err = pc.Controller.kubeclientset.AppsV1().Deployments(cspi.Namespace).Patch(existingDeployObj.Name, k8stypes.StrategicMergePatchType, strategicPatchData)
```

Comment:
As discussed we need to handle an upgrade scenario where CSPC-Operator is updated but not CSPI we will take in next PR's.